objective.py

```diff
@@ -19,12 +19,11 @@ def __init__(self):
     def set_data(self, C):
         self.C = C
 
+    def get_one_solution(self):
+        return np.eye((self.C.shape[1]))
+
     def compute(self, B):
-        D = transform_set(B, self.C)
-        n_matrices, n_features = self.C.shape[:2]
-        diagonals = np.diagonal(D, axis1=1, axis2=2)
-        logdet = -np.linalg.slogdet(B)[1]
-        return logdet + 0.5 * np.sum(np.log(diagonals)) / n_matrices
+        return np.exp(loss(self.C, B))
```

Comment:
@tomMoral I needed to exp the loss to make it work. Basically the loss value is negative here and without the exp it just crashes with a did not converge message.

Reply:
If you need a > 0 loss that is meaningful, you could compute sum logdet(C_i) in the get_data, and subtract this quantity when you compute the loss.